### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by rafaeljesus**
- [x] Get rid of glide in favor of dep
- [x] Extract retry and breaker configs to new structs
- [x] Create rabbus interpreter struct which implements/interpret rabbus defined interface contract
- [x] Expose `NewRabbusWithMaganedConn` useful for when users want to implement their own re-connection strategy

### CODE REVIEW

File: rabbus.go [hunk 1]
```diff
@@ -42,16 +44,29 @@ type Config struct {
...
+}
+
+// Retry config for the in memory retry mechanism
+type Retry struct {
 	// Attempts is the max number of retries on broker outages.
 	Attempts int
 	// Sleep is the sleep time of the retry mechanism.
 	Sleep time.Duration
+}
+
```
Comment (FelipeUmpierre) [inline rabbus.go]: Could you add a description about this struct?

File: README.md [hunk 2]
```diff
@@ -22,10 +22,12 @@ import (
...
-    Dsn      : "amqp://guest:guest@localhost:5672",
-    Attempts : 5,
-    Sleep     : time.Second * 2,
-    Durable  : true,
+    Dsn         : "amqp://guest:guest@localhost:5672",
+    Durable     : true,
+    Retry       : rabbus.Retry {
+      Attempts  : 5,
+      Sleep     : time.Second * 2,
+    },
```
Comment (FelipeUmpierre) [inline README.md]: Could you add the example of how to use `Breaker`?

File: rabbus.go [hunk 3]
```diff
@@ -237,24 +250,31 @@ func (r *rabbus) Listen(c ListenConfig) (chan ConsumerMessage, error) {
 }
 
 // Close attempt to close channel and connection.
-func (r *rabbus) Close() {
-	r.ch.Close()
-	r.conn.Close()
+func (ri *RabbusInterpreter) Close() (err error) {
+	if err = ri.ch.Close(); err != nil {
+		return err
```
Comment (FelipeUmpierre) [inline rabbus.go]: Shouldn't be just? https://play.golang.org/p/VKEF3WwA2v

```go
if err = ri.ch.Close(); err != nil {
     return
}
```

Comment (rafaeljesus) [reply rabbus.go]: Good catch!!

File: rabbus.go (hunk 1 above)
Comment (rafaeljesus) [reply rabbus.go]: Thx!

File: README.md (hunk 2 above)
Comment (rafaeljesus) [reply README.md]: Sure 👍

**Comment by FelipeUmpierre** [comment]
+1

**Pull request merged**